tasks/sync_repo_languages.py

```diff
@@ -0,0 +1,78 @@
+import logging
+
+from shared.celery_config import sync_repo_languages_task_name
+from shared.torngit.exceptions import TorngitError
+from sqlalchemy import String
+from sqlalchemy.orm.session import Session
+
+from app import celery_app
+from database.models.core import Repository
+from helpers.clock import get_utc_now
+from services.repository import get_repo_provider_service
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+BUNDLE_ANALYSIS_LANGUAGES = ["javascript", "typescript"]
+
+# We sync on repos that don't have the desired BA languages every 7 days
+# TODO: make this number also tweakable by infra
+REPOSITORY_LANGUAGE_SYNC_THRESHOLD = 7
```

Comment:
Does anyone know how to do this? Quickly looked through something like a settings or config object but couldn't find it

Reply:
Main reason for this value was for self-hosted customers to not run this very often so the infra team had more flexibility with these.

@trent-codecov, this is the PR to ping the providers to sync with the repos. TLDR, the sync will run once, then it will run every 7 days if the repo doesn't have js/ts, or anytime that the "syncing" button is pressed. Do you still want to add some custom flag to disable this for certain clients or have control on this threshold?

Reply:
We can forgo the flag for now. If we see rate limit issues, we can revisit. Thanks for taking the extra work to accommodate this.

Reply:
No problem! Know where to look if we see issues